x-pack/plugins/apm/server/lib/helpers/setup_request.ts

```diff
@@ -98,11 +99,14 @@ export async function setupRequest<TParams extends SetupRequestParams>(
       context,
       request,
     }),
-    ml: getMlSetup(
-      context.plugins.ml,
-      context.core.savedObjects.client,
-      request
-    ),
+    ml:
+      context.plugins.ml && isValidPlatinumLicense(context.licensing.license)
```

Comment:
Not sure we should check for the license here

Reply:
My thinking is we cannot use ML when platinum is disabled, so it might be easier to not set it up at all, and prevent us from having to do multiple license checks downstream.